dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/dto/ApiYARNApplicationDTO.java

```diff
@@ -16,146 +16,56 @@
-  private long allocatedMemorySeconds;
-
-  @JsonProperty(required = true)
-  private long allocatedVcoreSeconds;
+  private String clusterName;
 
-  @JsonProperty(required = true)
-  private Map<String, String> attributes;
+  /**
+   * DTO class for the official Cloudera API <a
+   * href="https://archive.cloudera.com/cm7/7.0.3/generic/jar/cm_api/apidocs/json_ApiYarnApplication.html">json</a>.
+   * JsonNode used to support schema evaluation.
+   */
+  private JsonNode apiYARNApplication;
 
-  public String getClusterName() {
-    return clusterName;
+  public ApiYARNApplicationDTO(JsonNode apiYARNApplication) {
+    this.apiYARNApplication = apiYARNApplication;
   }
 
+  @JsonIgnore
   public String getApplicationId() {
-    return applicationId;
-  }
-
-  public String getName() {
-    return name;
-  }
-
-  public String getState() {
-    return state;
-  }
-
-  public String getStartTime() {
-    return startTime;
-  }
-
-  public String getEndTime() {
-    return endTime;
-  }
-
-  public String getUser() {
-    return user;
-  }
-
-  public String getPool() {
-    return pool;
-  }
-
-  public List<String> getApplicationTags() {
-    return applicationTags;
-  }
-
-  public long getAllocatedMemorySeconds() {
-    return allocatedMemorySeconds;
-  }
-
-  public long getAllocatedVcoreSeconds() {
-    return allocatedVcoreSeconds;
+    if (apiYARNApplication == null) {
+      return null;
+    }
+    String applicationId = apiYARNApplication.at("/applicationId").asText();
+    return StringUtils.hasText(applicationId) ? applicationId : null;
   }
 
-  public Map<String, String> getAttributes() {
-    return attributes;
+  public String getClusterName() {
+    return clusterName;
   }
 
   public void setClusterName(String clusterName) {
     this.clusterName = clusterName;
   }
 
-  public void setApplicationId(String applicationId) {
-    this.applicationId = applicationId;
-  }
-
-  public void setName(String name) {
-    this.name = name;
-  }
-
-  public void setState(String state) {
-    this.state = state;
-  }
-
-  public void setStartTime(String startTime) {
-    this.startTime = startTime;
-  }
-
-  public void setEndTime(String endTime) {
-    this.endTime = endTime;
-  }
-
-  public void setUser(String user) {
-    this.user = user;
-  }
-
-  public void setPool(String pool) {
-    this.pool = pool;
-  }
-
-  public void setApplicationTags(List<String> applicationTags) {
-    this.applicationTags = applicationTags;
-  }
-
-  public void setAllocatedMemorySeconds(long allocatedMemorySeconds) {
-    this.allocatedMemorySeconds = allocatedMemorySeconds;
-  }
-
-  public void setAllocatedVcoreSeconds(long allocatedVcoreSeconds) {
-    this.allocatedVcoreSeconds = allocatedVcoreSeconds;
+  public JsonNode getApiYARNApplication() {
+    return apiYARNApplication;
   }
 
-  public void setAttributes(Map<String, String> attributes) {
-    this.attributes = attributes;
+  public void setApiYARNApplication(JsonNode apiYARNApplication) {
```

Comment:
It looks inconsistent a little bit: `setApiYARNApplication` in function name Api in camel case, but YARN in upper case. What about `setApiYarnApplication`. It looks more consistent.

Reply:
removed